viewing_party/party.py

```diff
@@ -1,23 +1,152 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        new_movie = None
+    else:
+        new_movie = {"title": title, "genre": genre, "rating": rating}
+    return new_movie
 
+def add_to_watched(user_data, movie):
+    watched_list = user_data["watched"]
+    watched_list.append(movie)
+    user_data["watched"] = watched_list  
+        
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    watchlist = user_data["watchlist"]
+    watchlist.append(movie)
+    user_data["watchlist"] = watchlist  
+
+    return user_data
+
+def watch_movie(user_data, movie):
+    # look up movie in watchlist
+    tracker_dict = None
+    for movie_dict in user_data["watchlist"]:
+        if movie == movie_dict["title"]:
+            add_to_watched(user_data, movie_dict)
+            tracker_dict = movie_dict
+    if tracker_dict:
+        user_data["watchlist"].remove(tracker_dict)
+
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data): 
+    sum = 0 
 
+    if len(user_data["watched"]) != 0:
+        for i in range(len(user_data["watched"])):
+            sum += user_data["watched"][i]["rating"]

```

Comment:
In this situation, I think it's better to use a `for` loop on each item in `user_data["watched"]`, rather than `i` over a range. I think that because we use `i` to access the `user_data`. I'd choose to use `i` if we needed to use it as a number.

Consider the following refactor, which passes the tests!

```python
        for movie in user_data["watched"]:
            sum += movie["rating"]
```